test/IntegrationTests/FileVectorizer/test_paginatedfilelist.py

```diff
@@ -0,0 +1,253 @@
+        logger.info("Executing test_paginated_list_files_token_without_org_role")
+
+        headers = {'Authorization': f'Bearer {create_token_without_org_role(self.org_id)}'}
+        url = f"{self.BASE_URL}/file/list?page=1&page_size=10"
+        logger.info(f"Sending GET request to {url}")
+
+        response = requests.get(url, headers=headers)
+        logger.info(f"Received response status code: {response.status_code}")
+
+        self.assertEqual(response.status_code, 403, "Expected status code 403 for missing org_role")
+        self.assertIn("detail", response.json(), "'detail' key not found in response")
+        self.assertEqual(response.json()["detail"], "Forbidden: Insufficient role", "Unexpected error message")
+
+        logger.info("Test completed successfully for test_paginated_list_files_token_without_org_role")
+
+    def test_paginated_list_files_token_without_org_id(self):
+        """Test the API with a token missing the org_id claim."""
+        logger.info("Executing test_paginated_list_files_token_without_org_id")
+
+        headers = {'Authorization': f'Bearer {create_token_without_org_id("org:admin")}'}
+        url = f"{self.BASE_URL}/file/list?page=1&page_size=10"
+        logger.info(f"Sending GET request to {url}")
+
+        response = requests.get(url, headers=headers)
+        logger.info(f"Received response status code: {response.status_code}")
+
+        self.assertEqual(response.status_code, 400, "Expected status code 400 for missing org_id")
+        self.assertIn("detail", response.json(), "'detail' key not found in response")
+        self.assertEqual(response.json()["detail"], "Org ID not found in token", "Unexpected error message")
+
+        logger.info("Test completed successfully for test_paginated_list_files_token_without_org_id")
+
+    def test_paginated_list_files_no_mapping_customer_guid(self):
+        """Test the API with an org_id that has no mapping to a customer_guid."""
+        logger.info("Executing test_paginated_list_files_no_mapping_customer_guid")
+
+        # Create a new org_id without mapping it to a customer_guid
+        org_id = "unmapped_org_id"
+        token = create_test_token(org_id=org_id, org_role="org:admin")
+        headers = {'Authorization': f'Bearer {token}'}
+        url = f"{self.BASE_URL}/file/list?page=1&page_size=10"
+        logger.info(f"Sending GET request to {url}")
+
+        response = requests.get(url, headers=headers)
+        logger.info(f"Received response status code: {response.status_code}")
+
+        self.assertEqual(response.status_code, 404, "Expected status code 404 for unmapped org_id")
+        self.assertIn("detail", response.json(), "'detail' key not found in response")
+        self.assertEqual(response.json()["detail"], "Invalid customer_guid provided", "Unexpected error message")
+
+        logger.info("Test completed successfully for test_paginated_list_files_no_mapping_customer_guid")
+
```

Comment:
1) Can we upload 50 files and then get status and check for different page sizes (10, 20, 30) and check whether the page numbers and the file status are arriving correctly. Check with Bharani - he has written similar testcases for tickets pagination - you can leverage same type of testcase writing

2) Can we delete few files like 25th file, 5th file ,42nd file and check when we retrieve the file status for different page sizes(10, 20, 30) , the deleted file data should not be present - If we have not wrriten delete file apis- then leave the (2) testcase